src/sinks/datadog_archives.rs

```diff
@@ -87,7 +87,7 @@ impl SinkBatchSettings for DatadogArchivesDefaultBatchSettings {
     const TIMEOUT_SECS: f64 = 900.0;
 }
 /// Configuration for the `datadog_archives` sink.
-#[configurable_component(sink)]
+#[configurable_component(sink("datadog_archives"))]
```

Comment:
I guess we were intentionally not `inventory::submit`ing this one?

Reply:
Hmmmmm, that's a good point.

Maybe I should drop the `sink(...)` for now and leave it out of the source enum... 🤔

Reply:
Fixed in 2d9bdb6.

I ended up dropping the `sink(...)` annotation and leading a TODO explaining why, as well as dropping it from the big sink enum.